### DISCUSSION TITLE
Could we get help with installing hslayers-ng

### DISCUSSION THREAD

**Discussion by SABR-13** [hslayers/hslayers-ng#4631] [Q&A]
Hi,

we are trying to install hslayers-ng along with hs-layers server to create our own web map service. Currently all we have setup is an empty remote ubuntu server. 

Our understanding currently is that we should
1) install hslayers-server
2) install hslayers-ng

What are the steps to get hslayers-ng running in order to get e.g. our own WMS layers visible on the map webpage? What web server is needed or is that included in hslayers-server or layman? Should we install layman and when? Should we install with Angular CLI or not?

Our main issue is that it is not clear what needs to be installed and in what order to get the service working. Any help would be appreciated. Even a list of what to install and when would be sufficient.

And if we want to create a development and production environment, is there something else we should do?

Thanks!

**Comment by jmacura** [comment]
Dear @SABR-13 , thanks for using our apps!

If you only have an empty Ubuntu server, you will also need to set up some web server like Apache or nginx, in order to publish webpages.
For hslayers-ng itself, you do not need to install Angular CLI separately, it will be installed together with hslayers-ng. https://github.com/hslayers/hslayers-ng/wiki/Getting-started page might be helpful.

If you intend to also publish custom geospatial data, you will then need to also install [Layman](https://github.com/LayerManager/layman), please refer to their README.

hslayers-server is only needed to connect hslayers-ng with Layman and to ensure remote WMS/WFS services are not blocked by CORS errors. It does not provide any web-server/publishing functionality.

Feel free to ask again if you need more help.

**Reply by SABR-13** [reply]
Thank you very much for a quick response! This clears things a lot already. We will start installing in the order you mentioned. We do need to avoid CORS errors so Layman and hslayers-server is also needed.

**Comment by SABR-13** [comment]
Hi again,

we have successfully installed and ran Layman (Docker demo / dev) on our remote server and had a look around, but we are not familiar with the framework. In what file location do we npm install hslayers-server and hslayers-ng to test them?

Installing hslayers-server currently gives us an path error with better sqlite 3 (code 1, parth node_modules/better-sqlite3, command failed)

We also could not get Wagtail to start its service due to a permission error and we believer this is why the Layman test client does not show a login page.

Thanks!

**Reply by jmacura** [reply]
Hello again,

as you mention Wagtail, it looks like you want to run the whole [hub4everybody](https://hub4everybody.com/) solution. In that case I recommend you to use the contacts listed there to get a support (as a paid service).

If you intend to use HSLayers + Layman to present only your data, you shall be fine installing just these two. In that case, you can install hslayers-ng anywhere you like. If you only want to run the bare hslayers map without any changes or additions, you may check [hslayers-ng-app](https://www.npmjs.com/package/hslayers-ng-app), which you can just insert into a web page and you are done. To make things work, you then set HsConfig "[useProxy](https://www.npmjs.com/package/hslayers-ng-app)" and "[shareServiceUrl](shareServiceUrl)" values so they point to the relevant hslayers-server endpoints. For setting up [hslayers-server](https://github.com/hslayers/hslayers-ng/tree/develop/projects/hslayers-server), please refer to the README there and ask if you stack somewhere.

Best regards!
